### PR TITLE
WF-63858 Added UG documentation of an API to find an item in Word document using its property

### PR DESCRIPTION
**Pull request by elizabethatienoo**
<table>
<tr>
<td>
Task Link
</td>
<td>
<a href="https://syncfusion.atlassian.net/browse/WF-63858">https://syncfusion.atlassian.net/browse/WF-63858</a>
</td>
</tr>
<tr>
<td>
Description
</td>
<td>
Add UG documentation of an API to find an item
</td>
</tr>
<tr>
<td>
Solution
</td>
<td>
Add UG documentation of an API to find an item
</td>
</tr>
<tr>
<td>
Content review confirmation
</td>
<td>
 Done
</td>
</tr>
</table>

### CODE REVIEW

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats-toc.html [hunk 1]
```diff
@@ -161,6 +161,9 @@
             <li>
                 <a href="/file-formats/DocIO/Working-with-Find-and-Replace">Working with Find and Replace</a>
             </li>
+            <li>
```
Comment (MohanaselvamJothi) [inline File-Formats-toc.html:164]: Can you please change into sublist item for Find and replace page

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 2]
```diff
@@ -0,0 +1,738 @@
+---
+title: Find item in Word document | DocIO | Syncfusion
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:2]: Find item in Word document in .NET Word library | Syncfusion

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 3]
```diff
@@ -0,0 +1,738 @@
...
+
+{% tabs %}
+
+{% highlight c# tabtitle="C#" %}
+//Load an existing Word document.
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    //Find all footnote and endnote by EntityType in Word document.
+    List<Entity> footNotes = document.FindAllItemsByProperty(EntityType.Footnote, null, null);
+    //Iterates the footnotes and remove the footnotes.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:305]: //Remove footnotes and endnotes

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 4]
```diff
@@ -0,0 +1,738 @@
...
+    for (int i = 0; i < footNotes.Count; i++)
+    {
+        WFootnote footnote = footNotes[i] as WFootnote;
+        footnote.OwnerParagraph.ChildEntities.Remove(footnote);
+    }
+
+    //Find all fields by FieldType.
+    List<Entity> fields = document.FindAllItemsByProperty(EntityType.Field, "FieldType",FieldType.FieldHyperlink.ToString());
+    //Iterates the fields
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:314]: Iterate the hyperlink field and change URL

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 5]
```diff
@@ -0,0 +1,738 @@
...
+{% tabs %}
+
+{% highlight c# tabtitle="C#" %}
+//Load an existing Word document.
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    string[] propertyNames = { "ContentControlProperties.Title", "ContentControlProperties.Tag" };
+    string[] propertyValues = { "CompanyName", "CompanyName" };
+
+    //Find all block content control by Title and Tag.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:483]: //Find all block content controls by Title and Tag.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 6]
```diff
@@ -0,0 +1,738 @@
...
+    for (int i = 0; i < blockContentControls.Count; i++)
+    {
+        BlockContentControl blockContentControl = blockContentControls[i] as BlockContentControl;
+        blockContentControl.OwnerTextBody.ChildEntities.Remove(blockContentControl);
+    }
+
+    propertyNames = new string[] { "ContentControlProperties.Title", "ContentControlProperties.Tag" };
+    propertyValues = new string[] { "Contact", "Contact" };
+
+    //Find all inline content control by Title and Tag.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:496]: //Find all inline content controls by Title and Tag.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 7]
```diff
@@ -0,0 +1,738 @@
...
+        inlineContentControl.OwnerParagraph.ChildEntities.Remove(inlineContentControl);
+    }
+
+    propertyNames = new string[] { "CharacterFormat.Bold", "CharacterFormat.Italic" };
+    propertyValues = new string[] { true.ToString(), true.ToString() };
+
+    //Find all bold and italic text.
+    List<Entity> textRanges = document.FindAllItemsByProperties(EntityType.TextRange,propertyNames,propertyValues);
+
+    //Iterates the textRanges and Remove the bold and italic
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Find-item-in-word-document.md:512]: //Iterates the textRanges and remove the bold and italic

File: File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md [hunk 8]
```diff
@@ -1059,4 +1059,8 @@ private static void IterateParagraph(ParagraphItemCollection paraItems)
 
 {% endtabs %}
 
-You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Word-document/Iterate-document-elements).
\ No newline at end of file
+You can download a complete working sample from [GitHub](https://github.com/SyncfusionExamples/DocIO-Examples/tree/main/Word-document/Iterate-document-elements).
+
+T> Passing null for both the property names and property values in above APIs, you can also find an item in a Word document without relying on any property.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md:1064]: Please move this tip to Find an item page

File: File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md [hunk 9]
```diff
@@ -1059,4 +1059,8 @@ private static void IterateParagraph(ParagraphItemCollection paraItems)
...
+
+T> If you wish to find an item in Word document rather than iterating through each element one by one, you can use finding the item functionality to achieve it.
```
Comment (MohanaselvamJothi) [inline File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md:1066]: Please add hyperlink "[finding the item functionality](https://help.syncfusion.com/find-item-in-word-document)"

**Review by MohanaselvamJothi — CHANGES_REQUESTED** [review]

File: File-Formats-toc.html [hunk 10]
```diff
@@ -161,6 +161,9 @@
...
+                <a href="/file-formats/DocIO/Find-item-in-word-document">Find-item-in-word-document</a>
```
[the remark this answers is not in the document]
Comment (elizabethatienoo) [reply File-Formats-toc.html]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+}
+//Download the helper files from the following link to save the stream as file and open the file for viewing in Xamarin platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+{% endhighlight %} 
+
+{% endtabs %}
+
+T> Passing null for both the property names and property values in above APIs, you can also find an item in a Word document without relying on any property.
+
+T> If you wish to find an item in Word document rather than iterating through each element one by one, you can use finding the item functionality to achieve it.
```
[the remark this answers is not in the document]
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats-toc.html (hunk 1 above)
Comment (elizabethatienoo) [reply File-Formats-toc.html]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 2 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 3 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 4 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 5 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 6 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md (hunk 9 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md]: Addressed

File: File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md (hunk 8 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Word-document/Iterating-Word-document-elements.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md (hunk 7 above)
Comment (elizabethatienoo) [reply File-Formats/DocIO/Find-item-in-word-document.md]: Addressed

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 12]
```diff
@@ -0,0 +1,740 @@
...
+        //Remove footnotes and endnotes.
+        for (int i = 0; i < footNotes.Count; i++)
+        {
+            WFootnote footnote = footNotes[i] as WFootnote;
+            footnote.OwnerParagraph.ChildEntities.Remove(footnote);
+        }
+
+        //Find all fields by FieldType.
+        List<Entity> fields = document.FindAllItemsByProperty(EntityType.Field, "FieldType",FieldType.FieldHyperlink.ToString());
+        //Iterate the hyperlink field and change URL.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:371]: //Iterate the hyperlink field and change the URL.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 13]
```diff
@@ -0,0 +1,740 @@
...
+        //Find all footnote and endnote by EntityType in Word document.
+        List<Entity> footNotes = document.FindAllItemsByProperty(EntityType.Footnote, null, null);
+        //Remove footnotes and endnotes.
+        for (int i = 0; i < footNotes.Count; i++)
+        {
+            WFootnote footnote = footNotes[i] as WFootnote;
+            footnote.OwnerParagraph.ChildEntities.Remove(footnote);
+        }
+
+        //Find all fields by FieldType.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:369]: //Find all the fields by FieldType.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 14]
```diff
@@ -0,0 +1,740 @@
...
+{% highlight c# tabtitle="UWP" %}
+//Open the file as Stream.
+using (Stream docStream = typeof(App).GetTypeInfo().Assembly.GetManifestResourceStream("Sample.Assets.Input.docx"))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Find all footnote and endnote by EntityType in Word document.
+        List<Entity> footNotes = document.FindAllItemsByProperty(EntityType.Footnote, null, null);
+        //Remove footnotes and endnotes.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:362]: //Remove the footnotes and endnotes.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 15]
```diff
@@ -0,0 +1,740 @@
...
+{% endhighlight %}
+
+{% highlight c# tabtitle="UWP" %}
+//Open the file as Stream.
+using (Stream docStream = typeof(App).GetTypeInfo().Assembly.GetManifestResourceStream("Sample.Assets.Input.docx"))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Find all footnote and endnote by EntityType in Word document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:360]: //Find all the footnotes and endnotes by the EntityType in the Word document.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 16]
```diff
@@ -0,0 +1,740 @@
...
+    }
+    //Save a Word document.
+    document.Save("Sample.docx", FormatType.Docx);
+}
+{% endhighlight %}
+
+{% highlight vb.net tabtitle="VB.NET" %}
+'Load an existing Word document.
+Using document As WordDocument = New WordDocument("Input.docx", FormatType.Docx)
+    'Find all footnote and endnote by EntityType in Word document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:331]: 'Find all the footnotes and endnotes by EntityType in the Word document.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 17]
```diff
@@ -0,0 +1,740 @@
...
+    //Find all footnote and endnote by EntityType in Word document.
+    List<Entity> footNotes = document.FindAllItemsByProperty(EntityType.Footnote, null, null);
+    //Remove footnotes and endnotes.
+    for (int i = 0; i < footNotes.Count; i++)
+    {
+        WFootnote footnote = footNotes[i] as WFootnote;
+        footnote.OwnerParagraph.ChildEntities.Remove(footnote);
+    }
+
+    //Find all fields by FieldType.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:312]: //Find all the fields by FieldType.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 18]
```diff
@@ -0,0 +1,740 @@
...
+
+The following code example illustrates how to find all the items in Word document based on one property.
+
+{% tabs %}
+
+{% highlight c# tabtitle="C#" %}
+//Load an existing Word document.
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    //Find all footnote and endnote by EntityType in Word document.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:303]: //Find all the footnotes and endnotes by EntityType in the Word document.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 19]
```diff
@@ -0,0 +1,740 @@
...
+}
+//Download the helper files from the following link to save the stream as file and open the file for viewing in Xamarin platform
+//https://help.syncfusion.com/file-formats/docio/create-word-document-in-xamarin#helper-files-for-xamarin
+{% endhighlight %} 
+
+{% endtabs %}
+
+## Find all items by property
+
+Using FindAllItemsByProperty API, find all the items in the Word document that has the specified property name and value.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:293]: Using the FindAllItemsByProperty API, find all the items in the Word document with the specified property name and value.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 20]
```diff
@@ -0,0 +1,740 @@
...
+{% highlight c# tabtitle="Xamarin" %}
+//Open the file as Stream.
+using (Stream docStream = typeof(App).GetTypeInfo().Assembly.GetManifestResourceStream("Sample.Assets.Input.docx"))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        string[] propertyNames = { "ChartType", "ChartTitle" };
+        string[] propertyValues = { OfficeChartType.Pie.ToString(), "Sales" };
+        //Find chart by ChartType and ChartTitle.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:265]: //Find the chart by ChartType and ChartTitle.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 21]
```diff
@@ -0,0 +1,740 @@
...
+        WChart chart = document.FindItemByProperties(EntityType.Chart, propertyNames, propertyValues) as WChart;
+        //Rename the ChartTitle.
+        if (chart != null)
+        chart.ChartTitle = "Sales Analysis";
+
+        propertyNames =  new string[] { "Title","Rows.Count" };
+        propertyValues =  new string[]{ "SupplierDetails","6" };
+        //Find table by Title and Rows Count
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:273]: //Find the table by Title and Rows Count.

File: File-Formats/DocIO/Find-item-in-word-document.md [hunk 22]
```diff
@@ -0,0 +1,740 @@
...
+
+{% tabs %}
+
+{% highlight c# tabtitle="C#" %}
+//Load an existing Word document.
+using (WordDocument document = new WordDocument("Input.docx", FormatType.Docx))
+{
+    //Find all footnote and endnote by EntityType in Word document.
+    List<Entity> footNotes = document.FindAllItemsByProperty(EntityType.Footnote, null, null);
+    //Remove footnotes and endnotes.
```
Comment (florenceliliana) [inline File-Formats/DocIO/Find-item-in-word-document.md:305]: //Remove the footnotes and endnotes.

**Review by florenceliliana — CHANGES_REQUESTED** [review]

**Review by MohanaselvamJothi — APPROVED** [review]

**Comment by elizabethatienoo** [comment]
Rebuild

**Review by MohanaselvamJothi — APPROVED** [review]

**Pull request merged**